### PR TITLE
Add feature flag for BNLP Affirm Afterpay payment method

### LINKED ISSUE #6034: [BNPL] Add Feature Flag for Development of BNPL providers: Affirm & Afterpay
### Description
Create a feature flag for the development of the integration of Affirm & Afterpay BNPL providers. 

### Acceptance criteria
1. The feature flag should be turned off by default, so we do not interfere with production code while the feature is under development
2. The feature flag can be turned on for development.
3. The feature flag should take into account the server flag implemented in https://github.com/Automattic/woocommerce-payments-server/pull/3344

### PR DESCRIPTION
**Pull request by KarlisJ**
Fixes #6034 

#### Changes proposed in this Pull Request
Adds a feature flag for BNLP Affirm and Afterpay eligibility. 

The flag is purely server driven and value comes from the account cache.

#### Testing instructions
Covered with unit tests

-------------------

- [ ] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6288` or branch name `add/6034-feature-flag-bnlp` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=add/6034-feature-flag-bnlp) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 9cc3134c0afc7c1beac6fba9e65c457bc01af5ca
- Build time: 2023-05-25 13:58:18 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: includes/class-wc-payments-features.php [hunk 1]
```diff
@@ -286,6 +286,14 @@ public static function is_simplify_deposits_ui_enabled(): bool {
 		return '1' === get_option( self::SIMPLIFY_DEPOSITS_UI_FLAG_NAME, '1' );
 	}
 
+	/**
+	 * Checks whether the BNLP Affirm Afterpay is enabled.
```
Comment (mgascam) [inline includes/class-wc-payments-features.php]: typo: BNPL
```suggestion
	 * Checks whether the BNPL Affirm Afterpay is enabled.
```

**Review by mgascam — APPROVED** [review]
LGTM!

Just found a small typo in one of the comments.

File: changelog/add-6034-feature-flag-bnlp [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
```
Comment (mgascam) [inline changelog/add-6034-feature-flag-bnlp]: @KarlisJ I'd mark this as `patch` and maybe use the `dev` type. This comment is in the context of this conversation in Slack:
p1683298938613009-slack-CGGCLBN58

**Review by mgascam — APPROVED** [review]
LGTM

**Pull request merged**